hoss/coordinate_utilities.py

```diff
@@ -124,26 +124,30 @@ def get_coordinate_variables(
 
 
 def get_row_col_sizes_from_coordinate_datasets(
-    lat_arr: np.ndarray,
-    lon_arr: np.ndarray,
+    lat_arr: np.ndarray, lon_arr: np.ndarray, dim_order_is_y_x: bool
 ) -> tuple[int, int]:
     """
     This function returns the row and column sizes of the coordinate datasets
 
     """
-    # ToDo - if the coordinates are 3D
-    if lat_arr.ndim > 1 and lon_arr.shape == lat_arr.shape:
-        col_size = lat_arr.shape[1]
-        row_size = lat_arr.shape[0]
+    # ToDo - if there is an override configuration
+    if lat_arr.ndim >= 2 and lon_arr.shape == lat_arr.shape:
+        col_size = lat_arr.shape[-1]
+        row_size = lat_arr.shape[-2]
```

Comment:
This is assuming that the last two dimensions of the array correspond to the spatial dimensions. While that is what the CF-Conventions recommend, it is not ubiquitously adhered to. Is this is safe assumption? I think that is an assumption that is worth capturing in the documentation string for the function.

(Using -1 and -2 as the indices here feels better than 0 and 1, and should make things work for the 3-D SMAP L3 data we have)

Reply:
Yes. Most of the SMAP L3 3D(SPL3FTP, SPL3FTP_E and SPL3FTA)  does follow that except SPL3SMP which is basically 2D except for a couple of 3D variables which may need an exception configuration.

Reply:
updated the doc string https://github.com/nasa/harmony-opendap-subsetter/pull/19/commits/e922df1288b04815191390566370a7281876ca54